runtime/src/bank.rs

```diff
@@ -2280,12 +2280,8 @@ impl Bank {
                 invalid_vote_keys.insert(vote_pubkey, InvalidCacheEntryReason::WrongOwner);
                 return None;
             }
-            let Ok(vote_state) = vote_account.vote_state().cloned() else {
```

Comment:
nit: in the `get_vote_account` closure above we now have a redundant check for vote owner and deserializability since that's exactly what `VoteAccount::try_from` is doing now. There are some other redundant checks scattered around for things like vote owner as well. Can be cleaned up in a follow up.

```rust
            let account = self.get_account_with_fixed_root(vote_pubkey)?;
            if account.owner() == &solana_vote_program
                && VoteState::deserialize(account.data()).is_ok()
            {
                vote_accounts_cache_miss_count.fetch_add(1, Relaxed);
            }
            VoteAccount::try_from(account).ok()
```

Reply:
This code has been dead in a long time I think. We check that VoteAccounts contains all vote accounts at snapshot load time. We have the `vote_accounts_cache_miss_count` metric that's never been non-zero anytime I've looked since I started this cursed work.